tests/test_filling_fields.py

```diff
@@ -36,405 +34,385 @@
 )
 
 
-def assert_not_raise(method, parameters, exception):
-    try:
-        method(*parameters)
-    except exception:
-        msg = "Exception %s not expected to be raised" % exception.__name__
-        raise AssertionError(msg)
-
-
 @pytest.fixture
 def person(db):
     return mommy.make('generic.Person')
 
 
-class FieldFillingTestCase(TestCase):
-
-    def setUp(self):
-        self.person = mommy.make(models.Person)
+@pytest.fixture()
+def custom_cfg():
+    yield None
+    if hasattr(settings, 'MOMMY_CUSTOM_FIELDS_GEN'):
+        delattr(settings, 'MOMMY_CUSTOM_FIELDS_GEN')
+    mommy.generators.add('tests.generic.fields.CustomFieldWithGenerator', None)
+    mommy.generators.add('django.db.models.fields.CharField', None)
 
 
-class FillingFromChoice(FieldFillingTestCase):
+class TestFillingFromChoice():
 
-    def test_if_gender_is_populated_from_choices(self):
+    def test_if_gender_is_populated_from_choices(self, person):
         from tests.generic.models import GENDER_CH
-        self.assertTrue(self.person.gender in map(lambda x: x[0], GENDER_CH))
+        person.gender in map(lambda x: x[0], GENDER_CH)
 
-    def test_if_oppucation_populated_from_choices(self):
+    def test_if_oppucation_populated_from_choices(self, person):
         from tests.generic.models import OCCUPATION_CHOCIES
         occupations = [item[0] for list in OCCUPATION_CHOCIES for item in list[1]]
-        self.assertTrue(self.person.occupation in occupations)
+        person.occupation in occupations
 
 
-class StringFieldsFilling(FieldFillingTestCase):
+class TestStringFieldsFilling():
```

Comment:
Shall we add tests for failing cases like wrong type or a name with a length bigger than the limit? (This question applies to the other tests too)

Reply:
I don't think so. I feel that by doing that we'd be testing Django instead of model mommy